x/marker/keeper/send_restrictions.go

```diff
@@ -0,0 +1,142 @@
+package keeper
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	attrTypes "github.com/provenance-io/provenance/x/attribute/types"
+	"github.com/provenance-io/provenance/x/marker/types"
+)
+
+const AddressHasAccessKey = "address_has_access"
+
+func (k Keeper) AllowMarkerSend(ctx sdk.Context, from, to, denom string) error {
+	if HasMarkerSendRestrictionBypass(ctx) {
+		return nil
+	}
+
+	caller, err := sdk.AccAddressFromBech32(from)
```

Comment:
is this the most efficient way to handle this or are we doing the address/bech32 conversion multiple times...

Reply:
From this point on there are no more conversions happening.  I was actually thinking about this while writing this area of code.  

For example, the SendCoin call from the sdk uses AccAddr while the multi send uses the bech32 string.  So, I opted to convert the SendCoin `to` and `from` to strings in my function signature.  I needed to convert the from address back to AccAddress for checking module account and transfer permissions.  Also, for the `from` address uses the bech32 string to look up the attributes. 
I was originally going to use the function signature `AllowMarkerSend(ctx sdk.Context, from sdkAccAddress, to sdkAccAddress, denom string)` instead of `AllowMarkerSend(ctx sdk.Context, from, to, denom string)`.  However, after seeing how the multi-send worked I kept it as strings.

Looking in the future with this PR https://github.com/cosmos/cosmos-sdk/pull/14224 the signature will have to change to something `(ctx sdk.Context, from sdk.AccAddress, to sdk.AccAddress, coins sdk.Coins)`.  In their PR I'm not seeing how they are handling the multi-send and their respective string addresses.  Also, the sdk.Coins part will change the impl a bit too.

If you think it is worth the effort at the moment, I could change the signature to the future sdk change.  This will need some re-working of the sdk and the sdk.Coins will require changes to the `AllowMarkerSend` to loop through the coins and ensure all of them pass.

Reply:
I am happy to hear that the area is being noted. 👍🏻